R/check-uses-dontrun.R

```diff
@@ -0,0 +1,80 @@
+#' Check if examples use `\dontrun{}`
+#'
+#' This check identifies functions in the package documentation that have
+#' examples using `\dontrun{}`. The use of `\dontrun{}` is discouraged by
+#' CRAN and should only be used the example really cannot be executed
+#' (e.g. because of missing additional software, missing API keys, ...) by
+#' the user. Instead use methods to run the examples conditionally, such as
+#' with the `@examplesIf()` roxygen tag.
+#'
+#' @param checks A 'pkgcheck' object with full \pkg{pkgstats} summary and
+#' \pkg{goodpractice} results.
+#' @return character vector of function names that have examples using `\dontrun{}`.
+#' @noRd
+pkgchk_uses_dontrun <- function(checks) {
+  rd_files <- list.files(
+    fs::path(
+      checks$pkg$path,
+      "man"
+    ),
+    pattern = "\\.Rd$",
+    full.names = TRUE
+  )
+
+  parsed_rds <- lapply(
+    rd_files,
+    tools::parse_Rd,
```

Comment:
This will need to be in `tryCatch()`, because `parse_Rd` can and does error on un-parseable files.

Reply:
Fixed here: https://github.com/ateucher/pkgcheck/blob/303af089d874602e6231e16500ede5982c5f7980/R/check-uses-dontrun.R#L79